doc/rest_api/snippet_sharing/api-description.apib

```diff
@@ -0,0 +1,587 @@
+            + `application` - Application for which the configuration is meant
+            + `scope` - Application internal scope
+            + `slug` - Entry specific unique slug
+    + filter (string, optional) - Only entries containing the search string in either their path, title, description, author or tags will be added to the output.
+    + filterby (enum[string], optional) - On what fields the filter should be applied to (i.e. what fields to involve in search)
+        + Default: `all`
+        + Members
+            + `all` - All fields: key, title, description, author and tags
+            + `key` - Only the key
+            + `title` - Only the title
+            + `description` - Only the description
+            + `author` - Only the authors name
+            + `tags` - Only tags
+
++ Response 200 (application/json)
+    + Attributes (Entry List)
+
+
+## Get details for entry [GET /database/{organization}/{application}/{scope}/{slug}{?raw}]
+
+Returns detailed information about the entry with the specified key (path). If the optional parameter `raw` is supplied, a raw configuration is returned instead of the normal response.
+
++ Parameters
+    + organization (string, required) - The `organization` who built the `application` for which the configuration of the entry is meant
+    + application (string, required) - The `application` to which the configuration of the entry belongs to
+    + scope (string, required) - The `application` specific `scope` to which the configuration of the entry belongs to
+    + slug (string, required) - The unique identifier of the requested entry
+    + raw (string, optional) - If this parameter is supplied, the API will attempt to return the requested configuration snippet in the given `raw` format
+
++ Response 200 (application/json)
+    + Attributes (object)
+        + data (object) - The actual details about the entry
+            + key (object) - The key and its separate parts
+                + full (string) - The full key (path) of the entry
+                + organization (string) - The organization of the application to which the snippet belongs to
+                + application (string) - The application to which the snippet belongs to
+                + scope (string) - The application internal scope for the configuration snippet
+                + slug (string) - The unique identification slug of the entry
+            + meta (object) - Additional information about the entry
+                + title (string) - The descriptive title of the entry
+                + description (string) - The extended description of the entry
+                + author (string) - The name of the creator of the entry
+                + created_at (number) - A timestamp of the creation date of the entry
+            + value (array[object])
+                + format (string) - A configuration format, e.g. **ini**
```

Comment:
> Is this needed? What is the separator between snippets? What happens if some formats fail?

Is it needed... definitely not, but still a nice to have in my opinion. Impact of it could be quite dramatically with lots of plugins enabled though, that's worth a discussion.
Each format of the snippet has its own object in an array of the response, consisting of format name and the snippet itself (plain string, escaping is all done by the json libs). As the snippet is stored in the key database as if it was a normal configuration, it is very unlikely that it would fail the conversion into another format (in my opninion this would mean the storage plugin in question is broken?!). This already worked in an earlier version of my service by the way.

> sounds great, is it possible to give a full example how this can be used (and not only "e.g. ini")

In the API description?

> The problem is we have some plugins that succeed with anything you give to them (e.g. line, ini). We would need proper auto-detection, e.g. the plugins give a score how likely it is that they should be used for a task. It will always be a somewhat messy heuristic but I have to admit that some kind of auto-detection is very useful (also for mounting).

That's the issue. Currently I'm really limited when it comes to plugins that I can use, because some of them just always succeed. Somehow I tend to asking the user what format his snippet is in, simply to avoid problems when trying the import for all plugins. But if the users enters a wrong format with a plugin that always succeeds, the same problem would arise...

Another issue could be that some plugins export with an absolute path, but the path from the REST service will make no sense for an import, because it has a special location for the snippets. I've absolutely no idea how to fix that (except by doing a stupid string-replace that will break the import on the other end). But somehow it isn't really my problem / fault, so ...